### PR TITLE
Handle the FSAL property in the embedded ERKN

### PR DESCRIPTION
**Pull request by eggrobin**
This looks far more elegant than it did in the SPRK/SRKN.

### CODE REVIEW

File: integrators/embedded_explicit_runge_kutta_nyström_integrator_body.hpp [hunk 1]
```diff
@@ -188,6 +209,11 @@ void EmbeddedExplicitRungeKuttaNyströmIntegrator<Position,
           adaptive_step_size.tolerance_to_error_ratio(h, error_estimate);
     } while (tolerance_to_error_ratio < 1.0);
 
+    if (first_same_as_last) {
+      std::swap(g.front(), g.back());
```
Comment (pleroy) [inline integrators/embedded_explicit_runge_kutta_nyström_integrator_body.hpp]: using std::swap;

File: integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp [hunk 2]
```diff
@@ -99,13 +121,24 @@ TEST_F(EmbeddedExplicitRungeKuttaNyströmIntegratorTest,
               AllOf(Ge(2E-3 * Metre / Second), Le(3E-3 * Metre / Second)));
   EXPECT_EQ(t_final, solution.back().time.value);
   EXPECT_EQ(steps_forward, solution.size());
-
+  EXPECT_EQ(
+      (1 + initial_rejections) * 4 +
```
Comment (pleroy) [inline integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp]: Move to the previous line.

File: integrators/embedded_explicit_runge_kutta_nyström_integrator_body.hpp (hunk 1 above)
Comment (eggrobin) [inline integrators/embedded_explicit_runge_kutta_nyström_integrator_body.hpp]: Done.

File: integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp (hunk 2 above)
Comment (eggrobin) [inline integrators/embedded_explicit_runge_kutta_nyström_integrator_test.cpp]: Done.

**Pull request merged**